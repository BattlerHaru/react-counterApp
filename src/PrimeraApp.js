// import React from "react";

// Para obligar a que se usen ciertos props se agrega lo siguiente
import PropTypes from "prop-types";

// Para evitar usar fragments se puede utilizar una etiqueta basia <>
// import { Fragment } from "react";

// Esto es un componente

// Para las props ya no es comun tenerlas asi, por lo que se utiliza la desestructuracion
// const PrimeraApp = (props) => {
const PrimeraApp = ({ saludo = "Hola Mundo", subtitulo }) => {
  //   const saludo = "Hola Mundo";
  //   const saludo = {
  //       nombre: 'test',
  //       edad: 34
  //   }

  return (
    <>
      <h1>{saludo}</h1>
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
      <p>{subtitulo}</p>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  subtitulo: "Soy un subtitulo",
};

export default PrimeraApp;
