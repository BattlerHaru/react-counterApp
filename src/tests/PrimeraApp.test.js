// import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";
import "@testing-library/jest-dom";

describe("Pruenas en <PrimeraApp />", () => {
  // se comenta la prueba por que utilizar esta lib es algo compleja, asi que utilizaremos enzyme
  //   test("Debe de mostar el mensaje: Hola, soy Goku", () => {
  //     const saludo = "Hola, soy Goku";

  //     // render recibe un componente a renderizar
  //     // const wrapper = render(<PrimeraApp saludo={saludo} />);
  //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

  //     // wrapper.getByText();

  //     expect(getByText(saludo)).toBeInTheDocument();
  //   });

  test("Debe de mostrar <PrimeraApp/> ", () => {
    const saludo = "Hola, soy Goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el subtitulo enviado por props ", () => {
    const saludo = "Hola, soy Goku";
    const subtitulo = "Soy un subtitulo";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );

    // El find es como el querryselector
    const textoParrafo = wrapper.find("p").text();

    expect(textoParrafo).toBe(subtitulo);
  });
});
