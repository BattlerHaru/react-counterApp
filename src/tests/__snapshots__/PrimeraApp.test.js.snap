// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruenas en <PrimeraApp /> Debe de mostrar <PrimeraApp/>  1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
